app/src/main/java/pl/skipcode/kotlinbaseappmvvm/feature/commons/ui/BaseActivity.kt

```diff
@@ -0,0 +1,106 @@
+package pl.skipcode.kotlinbaseappmvvm.feature.commons.ui
+
+import android.content.Intent
+import android.os.Bundle
+import android.os.Parcelable
+import android.widget.FrameLayout
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentTransaction
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.ViewModelProviders
+import dagger.android.AndroidInjection
+import pl.skipcode.kotlinbaseappmvvm.R
+import pl.skipcode.kotlinbaseappmvvm.feature.commons.BaseContract
+
+
+abstract class BaseActivity: AppCompatActivity(), BaseContract.ActivityView{
+
+    abstract val layoutId: Int
+    abstract val frameLayoutId: Int?
+
+    enum class ANIM {
+        FADE_IN, NONE
+    }
+
+    private var frameLayout: FrameLayout? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        AndroidInjection.inject(this)
+        super.onCreate(savedInstanceState)
+        setContentView(layoutId)
+
+        if(frameLayoutId != null) {
+            frameLayout = findViewById(frameLayoutId!!)
+        }
+    }
+
+    override fun <T> startActivity(
+            activityClass: Class<T>,
+            flags: List<Int>?,
+            parcelable: Pair<String, Parcelable>?) {
+        val intent = Intent(this, activityClass)
+        flags?.forEach { intent.addFlags(it) }
+        parcelable?.let { intent.putExtra(it.first, it.second) }
+        startActivity(intent)
+    }
+
+    protected inline fun <reified T : ViewModel> getViewModel(
+            viewModelFactory: ViewModelProvider.Factory
+    ): T = ViewModelProviders.of(this, viewModelFactory)[T::class.java]
+
+
+    private fun getAnim(animType: ANIM) : List<Int> {
+        return when (animType) {
+            ANIM.FADE_IN -> listOf(R.anim.fade_in, R.anim.fade_out,
+                    R.anim.fade_in, R.anim.fade_out)
+            ANIM.NONE -> listOf()
+        }
+    }
+
+    override fun setFragment(fragment: Fragment, animType: ANIM){
+        setFragmentExt(fragment, animType)
+    }
+
+    override fun addFragment(fragment: Fragment, animType: ANIM){
+        addFragmentExt(fragment, animType)
+    }
+
+    override fun replaceFragment(fragment: Fragment, animType: ANIM){
+        replaceFragmentExt(fragment, animType)
+    }
+
+    private fun AppCompatActivity.setFragmentExt(fragment: Fragment, animType: ANIM){
+        if (frameLayoutId == null) return
+
+        while (supportFragmentManager.backStackEntryCount > 0) {
+            supportFragmentManager.popBackStackImmediate()
+        }
+        supportFragmentManager.inTransaction ({ replace(frameLayoutId!!, fragment) }, animType)
+    }
+
+    private fun AppCompatActivity.addFragmentExt(fragment: Fragment, animType: ANIM){
+        if (frameLayoutId == null) return
```

Comment:
odwróć warunek -> if (frameLayoutId != null( {} i tam wsadź ciało tej metody, nie będziesz musiał używać !! do force-unwrapa nullablowego obiektu, generalnie używanie "!!" to jest zło i powinno się tego unikać jak ognia.

btw w kotlinie rzadko się używa if (foo != null). lepiej zrobić foo?.let { it.metoda() } tam wewnątrz wąsatych nawiasów it to jest to foo tylko już nie nullable